src/PrestaShopBundle/Controller/Admin/Sell/Catalog/ManufacturerController.php

```diff
@@ -65,15 +66,31 @@
 use PrestaShopBundle\Security\Annotation\AdminSecurity;
 use PrestaShopBundle\Security\Annotation\DemoRestricted;
 use PrestaShopBundle\Service\Grid\ResponseBuilder;
+use Symfony\Component\Filesystem\Filesystem;
 use Symfony\Component\HttpFoundation\RedirectResponse;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
 
 /**
  * Manages "Sell > Catalog > Brands & Suppliers > Brands" page
  */
 class ManufacturerController extends FrameworkBundleAdminController
 {
+    /**
+     * @var Filesystem
+     */
+    public $fs;
+
+    /**
+     * ManufacturerController constructor.
+     */
+    public function __construct()
+    {
+        parent::__construct();
+        $this->fs = new Filesystem();
```

Comment:
So far the FileSystem object is only used in the method `deleteCoverImageAction` , what's the point of instantiating it in the constructor? In most of the cases it will be instantiated needlessly.

Reply:
So ? Change is needed ? We can instanciate the FileSystem object directly in deleteCoverImageAction, and if the controller need to use object in other method in the futur the instanciation can be set in the constructor, right ?

Reply:
Yes, I'd say it's better to instantiate it in `deleteCoverImageAction` directly, since there is no dependency injection anyway.

Reply:
done

Reply:
thanks !